docs/guides/deploying/aws-datadog-monitoring.mdx

```diff
@@ -0,0 +1,207 @@
+---
+description: Instrument your Nitric application with Datadog for monitoring.
+tags:
+  - Monitoring
+  - AWS
+languages:
+  - typescript
+  - javascript
+published_at: 2024-11-26
+---
+
+# Instrumenting your Nitric application with Datadog
+
+This guide walks you through the steps to instrument your Nitric app with Datadog without changing any application code.
+Once you're done, you'll be able to monitor your application from Datadog's dashboard.
+
+<img
+  src="/docs/images/guides/aws-datadog-monitoring/datadog-dashboard.png"
+  style={{ maxWidth: 800, width: '100%' }}
+  alt="Screenshot of Datadog dashboard showing metrics"
+/>
+
+## Prerequisites
+
+- [Node.js](https://nodejs.org/en/download/)
+- The [Nitric CLI](/get-started/installation)
+- An account with [Datadog](https://www.datadoghq.com/)
+- An account with [AWS](https://aws.amazon.com/)
+
+## Getting started
+
+Begin by creating a new project using the Nitric TypeScript template and installing dependencies:
+
+```bash
+nitric new my-api ts-starter
+cd my-api
+npm install
+```
+
+## Add Datadog libraries to your container image
+
+Nitric applications deploy as containerized applications, you'll need to include the Datadog Lambda and tracing libraries in your container image:
+
+```bash
+npm install datadog-lambda-js dd-trace
+```
+
+## Add the Datadog Lambda extension
+
+To enable Datadog monitoring, add the Datadog Lambda Extension to your container image by modifying your `node.dockerfile`. Add the following lines to the end of your dockerfile, just before you define your entrypoint `ENTRYPOINT ["node", "lib/index.js"]`.
+
+1. **Fetch and copy the Datadog Lambda Extension**  
+   Retrieve the Lambda Extension from Amazon's Elastic Container Registry (ECR) and copy its contents into the `/opt/` directory of your image. This extension runs as a sidecar process, collecting and forwarding telemetry data to Datadog.
+
+   ```dockerfile
+   COPY --from=public.ecr.aws/datadog/lambda-extension:latest-alpine /opt/. /opt/
+   ```
+
+2. **Remove the default Datadog handler file**  
+   The default `handler.js` file from `datadog-lambda-js` might contain unnecessary boilerplate functionality.
+
+   ```dockerfile
+   RUN rm node_modules/datadog-lambda-js/dist/handler.js
+   ```
```

Comment:
This step may not be that useful. It won't shrink the size of the container since the file is still in the earlier layer. You'd want to add it to the end of the `yarn install` line to really remove it.